uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/request-modal-details-toil.controller.js

```diff
@@ -264,8 +262,9 @@ define([
      * @return {Promise}
      */
     function onDateChangeExtended () {
-      return (canCalculateExpiryDate() && calculateToilExpiryDate().catch($q.resolve)) ||
-        $q.resolve();
+      return canCalculateExpiryDate()
```

Comment:
This and `setDaysSelectionModeExtended` looks almost same except the `.catch($q.resolve)` part. Is it intentional? if not lets reuse `setDaysSelectionModeExtended`.

Reply:
The catch was first introduced here:

https://github.com/civicrm/civihr/commit/f8d0e9123d5a13093b331f74ff1d8a7482c1f0bb

I think this was added because `getReferenceDate` could return a rejected promise in case the reference date (From or To date depending on the multiple days selected).

For now I'll make them both use `.catch($q.resolve)` and make a note of refactoring `getReferenceDate` since that function needs attention but I have overextended my solution here.